src/lib.rs

```diff
@@ -54,17 +54,13 @@ where
     fn new(mut iter: I) -> Self {
         // Prepare the indices.
         let mut indices = [0; N];
-        for i in 0..N {
-            indices[i] = i;
+        for (i, index) in indices.iter_mut().enumerate() {
+            *index = i;
```

Comment:
Had to read this a few times over, but it's the same thing as before right? Not sure this is clearer than before

Reply:
I agree that it's somewhat less readable. I got warnings from clippy for using indexing instead of an interator. Maybe it's time to turn them off?